contracts/chain-adapters/Arbitrum_RescueAdapter.sol

```diff
@@ -0,0 +1,127 @@
+}
+
+/**
+ * @notice Meant to copy the Arbitrum_Adapter exactly in how it sends L1 --> L2 messages but is designed only to be
+ * used by the owner of the HubPool to retrieve ETH held by its aliased address on L2. This ETH builds up because
+ * `relayTokens` calls `l1ERC20GatewayRouter.outboundTransfer` which does not allow the caller to specify an L2 refund
+ * address the same way that `l1Inbox.createRetryableTicket` does. This means that the alias address of the caller, the
+ * HubPool in this case, receives ETH on L2. This Adapter can be used to send messages to Arbitrum specifically to send
+ * transactions as if called by the aliased HubPool address.
+ */
+
+// solhint-disable-next-line contract-name-camelcase
+contract Arbitrum_RescueAdapter is AdapterInterface {
+    using SafeERC20 for IERC20;
+
+    // Amount of ETH allocated to pay for the base submission fee. The base submission fee is a parameter unique to
+    // retryable transactions; the user is charged the base submission fee to cover the storage costs of keeping their
+    // ticket’s calldata in the retry buffer. (current base submission fee is queryable via
+    // ArbRetryableTx.getSubmissionPrice). ArbRetryableTicket precompile interface exists at L2 address
+    // 0x000000000000000000000000000000000000006E.
+    uint256 public immutable l2MaxSubmissionCost = 0.01e18;
+
+    // L2 Gas price bid for immediate L2 execution attempt (queryable via standard eth*gasPrice RPC)
+    uint256 public immutable l2GasPrice = 150e9;
+
+    // Gas limit for immediate L2 execution attempt (can be estimated via NodeInterface.estimateRetryableTicket).
+    // NodeInterface precompile interface exists at L2 address 0x00000000000000000000000000000000000000C8
+    uint32 public immutable l2GasLimit = 160_000;
+
+    // This address on L2 receives extra ETH that is left over after relaying a message via the inbox.
+    address public immutable l2RefundL2Address;
+
+    // L1 HubPool address aliased on L2: https://github.com/OffchainLabs/arbitrum/blob/master/docs/L1_L2_Messages.md#address-aliasing
+    address public immutable aliasedL2HubPoolAddress = 0xd297fA914353c44B2e33EBE05F21846f1048CFeB;
+
+    ArbitrumL1InboxLike public immutable l1Inbox;
+
+    /**
+     * @notice Constructs new Adapter.
+     * @param _l1ArbitrumInbox Inbox helper contract to send messages to Arbitrum.
+     */
+    constructor(ArbitrumL1InboxLike _l1ArbitrumInbox) {
+        l1Inbox = _l1ArbitrumInbox;
+
+        l2RefundL2Address = msg.sender;
+    }
+
+    /**
+     * @notice Send cross-chain message to aliased hub pool address on Arbitrum.
+     * @notice This contract must hold at least getL1CallValue() amount of ETH to send a message via the Inbox
+     * successfully, or the message will get stuck.
+     * @param message Data to send to aliased hub pool.
+     */
+    function relayMessage(address, bytes memory message) external payable override {
```

Comment:
We might want to enforce some restrictions on the length or type of data in message to ensure that if this is ever called by the HubPool to send roots to arbitrum that it reverts.

For instance, if we encode the ETH amount in the data portion, we probably want to ensure that the length is exactly long enough to store a uint256 and no more/no less.